{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "from sklearn.decomposition import TruncatedSVD\n",
    "from Processing_Functions import remove_redun, map_IDs"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Import National Institute of Health Genomic Data Commons (GDA) and Human Protein Atlas (HPA) Data "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ensembl</th>\n",
       "      <th>nih_ssm_in_cohort</th>\n",
       "      <th>nih_ssm_across_gdc</th>\n",
       "      <th>nih_cnv_gain</th>\n",
       "      <th>nih_cnv_loss</th>\n",
       "      <th>nih_tot_mutations</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000147481</td>\n",
       "      <td>0.144621</td>\n",
       "      <td>0.028155</td>\n",
       "      <td>0.076023</td>\n",
       "      <td>0.056530</td>\n",
       "      <td>92</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000105877</td>\n",
       "      <td>0.144621</td>\n",
       "      <td>0.084772</td>\n",
       "      <td>0.074074</td>\n",
       "      <td>0.007797</td>\n",
       "      <td>105</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000188107</td>\n",
       "      <td>0.144621</td>\n",
       "      <td>0.068585</td>\n",
       "      <td>0.111111</td>\n",
       "      <td>0.056530</td>\n",
       "      <td>95</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000125414</td>\n",
       "      <td>0.144621</td>\n",
       "      <td>0.062601</td>\n",
       "      <td>0.005848</td>\n",
       "      <td>0.042885</td>\n",
       "      <td>95</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000009694</td>\n",
       "      <td>0.144621</td>\n",
       "      <td>0.071653</td>\n",
       "      <td>0.052632</td>\n",
       "      <td>0.025341</td>\n",
       "      <td>110</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21087</th>\n",
       "      <td>ENSG00000146648</td>\n",
       "      <td>0.146384</td>\n",
       "      <td>0.052244</td>\n",
       "      <td>0.105263</td>\n",
       "      <td>0.019493</td>\n",
       "      <td>51</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21088</th>\n",
       "      <td>ENSG00000133958</td>\n",
       "      <td>0.146384</td>\n",
       "      <td>0.059916</td>\n",
       "      <td>0.025341</td>\n",
       "      <td>0.046784</td>\n",
       "      <td>110</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21089</th>\n",
       "      <td>ENSG00000134376</td>\n",
       "      <td>0.146384</td>\n",
       "      <td>0.056617</td>\n",
       "      <td>0.085770</td>\n",
       "      <td>0.007797</td>\n",
       "      <td>93</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21090</th>\n",
       "      <td>ENSG00000109061</td>\n",
       "      <td>0.146384</td>\n",
       "      <td>0.057691</td>\n",
       "      <td>0.009747</td>\n",
       "      <td>0.040936</td>\n",
       "      <td>93</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21091</th>\n",
       "      <td>ENSG00000134516</td>\n",
       "      <td>0.146384</td>\n",
       "      <td>0.054622</td>\n",
       "      <td>0.066277</td>\n",
       "      <td>0.033138</td>\n",
       "      <td>103</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>21092 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               ensembl  nih_ssm_in_cohort  nih_ssm_across_gdc  nih_cnv_gain  \\\n",
       "0      ENSG00000147481           0.144621            0.028155      0.076023   \n",
       "1      ENSG00000105877           0.144621            0.084772      0.074074   \n",
       "2      ENSG00000188107           0.144621            0.068585      0.111111   \n",
       "3      ENSG00000125414           0.144621            0.062601      0.005848   \n",
       "4      ENSG00000009694           0.144621            0.071653      0.052632   \n",
       "...                ...                ...                 ...           ...   \n",
       "21087  ENSG00000146648           0.146384            0.052244      0.105263   \n",
       "21088  ENSG00000133958           0.146384            0.059916      0.025341   \n",
       "21089  ENSG00000134376           0.146384            0.056617      0.085770   \n",
       "21090  ENSG00000109061           0.146384            0.057691      0.009747   \n",
       "21091  ENSG00000134516           0.146384            0.054622      0.066277   \n",
       "\n",
       "       nih_cnv_loss  nih_tot_mutations  \n",
       "0          0.056530                 92  \n",
       "1          0.007797                105  \n",
       "2          0.056530                 95  \n",
       "3          0.042885                 95  \n",
       "4          0.025341                110  \n",
       "...             ...                ...  \n",
       "21087      0.019493                 51  \n",
       "21088      0.046784                110  \n",
       "21089      0.007797                 93  \n",
       "21090      0.040936                 93  \n",
       "21091      0.033138                103  \n",
       "\n",
       "[21092 rows x 6 columns]"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Process the gdc dataset to include only desired columns and scores\n",
    "gdc = pd.read_csv(\"../data/raw_data/gdc_luad_genes.csv\")\n",
    "\n",
    "featureNameColumns = ['# SSM Affected Cases in Cohort', '# CNV Gain', '# CNV Loss']\n",
    "for i in featureNameColumns:\n",
    "    gdc[['1', '2', '3', '4']] = gdc[i].replace({',':''}, regex=True).str.split(' ', 3, expand=True)\n",
    "    gdc[i] = gdc['1'].astype(float)/gdc['3'].astype(float)\n",
    "    gdc = gdc.drop(['1', '2', '3', '4'], axis=1)\n",
    "\n",
    "gdc.drop(['Symbol', 'Name', 'Cytoband', 'Type', 'Annotations', 'Survival'], axis=1, inplace=True)\n",
    "\n",
    "gdc[['1', '2', '3']] = gdc['# SSM Affected Cases Across the GDC'].replace({',':''}, regex=True).str.split(' ', 2, expand=True)\n",
    "gdc['# SSM Affected Cases Across the GDC'] = gdc['1'].astype(float)/gdc['3'].astype(float)\n",
    "gdc = gdc.drop(['1', '2', '3'], axis=1)\n",
    "\n",
    "gdc = gdc.rename({'# SSM Affected Cases in Cohort': 'nih_ssm_in_cohort', '# SSM Affected Cases Across the GDC':'nih_ssm_across_gdc',\n",
    "    '# CNV Gain':'nih_cnv_gain', '# CNV Loss':'nih_cnv_loss', 'Gene ID':'ensembl', '# Mutations':'nih_tot_mutations'}, axis=1)\n",
    "\n",
    "gdc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\cbyle\\AppData\\Local\\Temp/ipykernel_7364/788959935.py:21: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  hpa_features[\"Tissue RNA - lung [NX]\"] = (hpa_features[\"Tissue RNA - lung [NX]\"] - hpa_features[\"Tissue RNA - lung [NX]\"].mean()) / hpa_features[\"Tissue RNA - lung [NX]\"].std()\n",
      "C:\\Users\\cbyle\\AppData\\Local\\Temp/ipykernel_7364/788959935.py:23: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  hpa_features[col] = (hpa_features[col] - hpa_features[col].mean()) / hpa_features[col].std()\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ensembl</th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>...</th>\n",
       "      <th>90</th>\n",
       "      <th>91</th>\n",
       "      <th>92</th>\n",
       "      <th>93</th>\n",
       "      <th>94</th>\n",
       "      <th>95</th>\n",
       "      <th>96</th>\n",
       "      <th>97</th>\n",
       "      <th>98</th>\n",
       "      <th>99</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000175899</td>\n",
       "      <td>0.443505</td>\n",
       "      <td>0.313574</td>\n",
       "      <td>0.430307</td>\n",
       "      <td>-0.285337</td>\n",
       "      <td>-0.351412</td>\n",
       "      <td>1.545059</td>\n",
       "      <td>-0.450936</td>\n",
       "      <td>-0.444200</td>\n",
       "      <td>-0.116774</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.018373</td>\n",
       "      <td>0.094550</td>\n",
       "      <td>0.009357</td>\n",
       "      <td>-0.039162</td>\n",
       "      <td>0.032721</td>\n",
       "      <td>-0.079607</td>\n",
       "      <td>0.503267</td>\n",
       "      <td>0.576245</td>\n",
       "      <td>0.444453</td>\n",
       "      <td>-0.574186</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000128274</td>\n",
       "      <td>0.532289</td>\n",
       "      <td>0.789331</td>\n",
       "      <td>-0.122159</td>\n",
       "      <td>0.019000</td>\n",
       "      <td>1.427162</td>\n",
       "      <td>-0.065659</td>\n",
       "      <td>0.001669</td>\n",
       "      <td>0.386983</td>\n",
       "      <td>-0.028900</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.007169</td>\n",
       "      <td>0.257972</td>\n",
       "      <td>-0.085797</td>\n",
       "      <td>0.044028</td>\n",
       "      <td>0.166343</td>\n",
       "      <td>-0.110665</td>\n",
       "      <td>0.140087</td>\n",
       "      <td>-0.089804</td>\n",
       "      <td>0.000472</td>\n",
       "      <td>0.025348</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000094914</td>\n",
       "      <td>1.605229</td>\n",
       "      <td>0.936681</td>\n",
       "      <td>0.345886</td>\n",
       "      <td>0.183699</td>\n",
       "      <td>-1.001764</td>\n",
       "      <td>-0.846346</td>\n",
       "      <td>0.550995</td>\n",
       "      <td>0.368205</td>\n",
       "      <td>0.202731</td>\n",
       "      <td>...</td>\n",
       "      <td>0.339739</td>\n",
       "      <td>0.019335</td>\n",
       "      <td>-0.186211</td>\n",
       "      <td>-0.270075</td>\n",
       "      <td>-0.205805</td>\n",
       "      <td>0.027753</td>\n",
       "      <td>-0.048839</td>\n",
       "      <td>-0.054099</td>\n",
       "      <td>-0.007369</td>\n",
       "      <td>0.078670</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000081760</td>\n",
       "      <td>0.467196</td>\n",
       "      <td>0.690415</td>\n",
       "      <td>-0.066723</td>\n",
       "      <td>0.335478</td>\n",
       "      <td>1.017124</td>\n",
       "      <td>0.029579</td>\n",
       "      <td>-0.131714</td>\n",
       "      <td>0.212387</td>\n",
       "      <td>0.075120</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.102811</td>\n",
       "      <td>-0.352452</td>\n",
       "      <td>0.199631</td>\n",
       "      <td>-0.049215</td>\n",
       "      <td>-0.092627</td>\n",
       "      <td>0.010099</td>\n",
       "      <td>0.120493</td>\n",
       "      <td>0.138474</td>\n",
       "      <td>-0.167452</td>\n",
       "      <td>0.051508</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000114771</td>\n",
       "      <td>0.956718</td>\n",
       "      <td>0.202132</td>\n",
       "      <td>-0.490047</td>\n",
       "      <td>-0.524460</td>\n",
       "      <td>0.467132</td>\n",
       "      <td>-0.207775</td>\n",
       "      <td>-0.494652</td>\n",
       "      <td>0.192615</td>\n",
       "      <td>0.077836</td>\n",
       "      <td>...</td>\n",
       "      <td>0.003828</td>\n",
       "      <td>-0.002623</td>\n",
       "      <td>0.002410</td>\n",
       "      <td>0.000992</td>\n",
       "      <td>-0.009565</td>\n",
       "      <td>-0.015291</td>\n",
       "      <td>0.011153</td>\n",
       "      <td>-0.007147</td>\n",
       "      <td>-0.007970</td>\n",
       "      <td>0.007936</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15008</th>\n",
       "      <td>ENSG00000198455</td>\n",
       "      <td>1.436813</td>\n",
       "      <td>-1.455513</td>\n",
       "      <td>0.510199</td>\n",
       "      <td>0.224208</td>\n",
       "      <td>0.323736</td>\n",
       "      <td>-0.171141</td>\n",
       "      <td>-0.060697</td>\n",
       "      <td>0.088626</td>\n",
       "      <td>0.103071</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.038890</td>\n",
       "      <td>0.041711</td>\n",
       "      <td>-0.017337</td>\n",
       "      <td>-0.032425</td>\n",
       "      <td>-0.000801</td>\n",
       "      <td>0.011581</td>\n",
       "      <td>-0.021367</td>\n",
       "      <td>0.014229</td>\n",
       "      <td>-0.007050</td>\n",
       "      <td>0.008374</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15009</th>\n",
       "      <td>ENSG00000070476</td>\n",
       "      <td>1.140170</td>\n",
       "      <td>-1.162868</td>\n",
       "      <td>0.551409</td>\n",
       "      <td>0.127004</td>\n",
       "      <td>0.119115</td>\n",
       "      <td>-0.120434</td>\n",
       "      <td>-0.421260</td>\n",
       "      <td>0.798186</td>\n",
       "      <td>0.041917</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.029933</td>\n",
       "      <td>0.029976</td>\n",
       "      <td>-0.026192</td>\n",
       "      <td>-0.024701</td>\n",
       "      <td>0.009399</td>\n",
       "      <td>0.022374</td>\n",
       "      <td>-0.019030</td>\n",
       "      <td>0.024306</td>\n",
       "      <td>-0.011823</td>\n",
       "      <td>0.017133</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15010</th>\n",
       "      <td>ENSG00000162378</td>\n",
       "      <td>0.782655</td>\n",
       "      <td>-0.057857</td>\n",
       "      <td>-0.397211</td>\n",
       "      <td>0.017498</td>\n",
       "      <td>-0.135123</td>\n",
       "      <td>-0.248882</td>\n",
       "      <td>-0.438176</td>\n",
       "      <td>0.113363</td>\n",
       "      <td>-0.295955</td>\n",
       "      <td>...</td>\n",
       "      <td>0.010618</td>\n",
       "      <td>0.012477</td>\n",
       "      <td>-0.010456</td>\n",
       "      <td>0.005255</td>\n",
       "      <td>0.017619</td>\n",
       "      <td>0.049873</td>\n",
       "      <td>0.018236</td>\n",
       "      <td>-0.051592</td>\n",
       "      <td>0.049855</td>\n",
       "      <td>-0.009883</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15011</th>\n",
       "      <td>ENSG00000159840</td>\n",
       "      <td>1.042306</td>\n",
       "      <td>0.183392</td>\n",
       "      <td>-0.327697</td>\n",
       "      <td>0.065425</td>\n",
       "      <td>-0.459893</td>\n",
       "      <td>0.507171</td>\n",
       "      <td>-0.610148</td>\n",
       "      <td>0.067066</td>\n",
       "      <td>0.368216</td>\n",
       "      <td>...</td>\n",
       "      <td>0.318299</td>\n",
       "      <td>0.066721</td>\n",
       "      <td>0.485114</td>\n",
       "      <td>-0.249986</td>\n",
       "      <td>0.321037</td>\n",
       "      <td>-0.002111</td>\n",
       "      <td>-0.092111</td>\n",
       "      <td>-0.032771</td>\n",
       "      <td>0.033117</td>\n",
       "      <td>-0.052496</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15012</th>\n",
       "      <td>ENSG00000074755</td>\n",
       "      <td>0.553160</td>\n",
       "      <td>0.068709</td>\n",
       "      <td>0.013208</td>\n",
       "      <td>0.760541</td>\n",
       "      <td>0.592771</td>\n",
       "      <td>-0.094544</td>\n",
       "      <td>0.326136</td>\n",
       "      <td>-0.636445</td>\n",
       "      <td>0.047971</td>\n",
       "      <td>...</td>\n",
       "      <td>0.005374</td>\n",
       "      <td>0.008867</td>\n",
       "      <td>-0.009764</td>\n",
       "      <td>-0.004633</td>\n",
       "      <td>0.006748</td>\n",
       "      <td>-0.017633</td>\n",
       "      <td>-0.008296</td>\n",
       "      <td>0.002019</td>\n",
       "      <td>0.006353</td>\n",
       "      <td>-0.002431</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>15013 rows × 101 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               ensembl         0         1         2         3         4  \\\n",
       "0      ENSG00000175899  0.443505  0.313574  0.430307 -0.285337 -0.351412   \n",
       "1      ENSG00000128274  0.532289  0.789331 -0.122159  0.019000  1.427162   \n",
       "2      ENSG00000094914  1.605229  0.936681  0.345886  0.183699 -1.001764   \n",
       "3      ENSG00000081760  0.467196  0.690415 -0.066723  0.335478  1.017124   \n",
       "4      ENSG00000114771  0.956718  0.202132 -0.490047 -0.524460  0.467132   \n",
       "...                ...       ...       ...       ...       ...       ...   \n",
       "15008  ENSG00000198455  1.436813 -1.455513  0.510199  0.224208  0.323736   \n",
       "15009  ENSG00000070476  1.140170 -1.162868  0.551409  0.127004  0.119115   \n",
       "15010  ENSG00000162378  0.782655 -0.057857 -0.397211  0.017498 -0.135123   \n",
       "15011  ENSG00000159840  1.042306  0.183392 -0.327697  0.065425 -0.459893   \n",
       "15012  ENSG00000074755  0.553160  0.068709  0.013208  0.760541  0.592771   \n",
       "\n",
       "              5         6         7         8  ...        90        91  \\\n",
       "0      1.545059 -0.450936 -0.444200 -0.116774  ... -0.018373  0.094550   \n",
       "1     -0.065659  0.001669  0.386983 -0.028900  ... -0.007169  0.257972   \n",
       "2     -0.846346  0.550995  0.368205  0.202731  ...  0.339739  0.019335   \n",
       "3      0.029579 -0.131714  0.212387  0.075120  ... -0.102811 -0.352452   \n",
       "4     -0.207775 -0.494652  0.192615  0.077836  ...  0.003828 -0.002623   \n",
       "...         ...       ...       ...       ...  ...       ...       ...   \n",
       "15008 -0.171141 -0.060697  0.088626  0.103071  ... -0.038890  0.041711   \n",
       "15009 -0.120434 -0.421260  0.798186  0.041917  ... -0.029933  0.029976   \n",
       "15010 -0.248882 -0.438176  0.113363 -0.295955  ...  0.010618  0.012477   \n",
       "15011  0.507171 -0.610148  0.067066  0.368216  ...  0.318299  0.066721   \n",
       "15012 -0.094544  0.326136 -0.636445  0.047971  ...  0.005374  0.008867   \n",
       "\n",
       "             92        93        94        95        96        97        98  \\\n",
       "0      0.009357 -0.039162  0.032721 -0.079607  0.503267  0.576245  0.444453   \n",
       "1     -0.085797  0.044028  0.166343 -0.110665  0.140087 -0.089804  0.000472   \n",
       "2     -0.186211 -0.270075 -0.205805  0.027753 -0.048839 -0.054099 -0.007369   \n",
       "3      0.199631 -0.049215 -0.092627  0.010099  0.120493  0.138474 -0.167452   \n",
       "4      0.002410  0.000992 -0.009565 -0.015291  0.011153 -0.007147 -0.007970   \n",
       "...         ...       ...       ...       ...       ...       ...       ...   \n",
       "15008 -0.017337 -0.032425 -0.000801  0.011581 -0.021367  0.014229 -0.007050   \n",
       "15009 -0.026192 -0.024701  0.009399  0.022374 -0.019030  0.024306 -0.011823   \n",
       "15010 -0.010456  0.005255  0.017619  0.049873  0.018236 -0.051592  0.049855   \n",
       "15011  0.485114 -0.249986  0.321037 -0.002111 -0.092111 -0.032771  0.033117   \n",
       "15012 -0.009764 -0.004633  0.006748 -0.017633 -0.008296  0.002019  0.006353   \n",
       "\n",
       "             99  \n",
       "0     -0.574186  \n",
       "1      0.025348  \n",
       "2      0.078670  \n",
       "3      0.051508  \n",
       "4      0.007936  \n",
       "...         ...  \n",
       "15008  0.008374  \n",
       "15009  0.017133  \n",
       "15010 -0.009883  \n",
       "15011 -0.052496  \n",
       "15012 -0.002431  \n",
       "\n",
       "[15013 rows x 101 columns]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hpa = pd.read_csv('../data/hpa_gene_features.tsv', sep='\\t').drop_duplicates(subset='Gene')\n",
    "\n",
    "identifiers = [\n",
    "    \"Gene\",\n",
    "    \"Ensembl\"\n",
    "]\n",
    "discrete_features = [\n",
    "    \"Protein class\",\n",
    "    \"Biological process\",\n",
    "    \"Molecular function\",\n",
    "    \"Disease involvement\",\n",
    "    \"Subcellular location\",\n",
    "]\n",
    "continuous_features = [\n",
    "    \"Tissue RNA - lung [NX]\",\n",
    "    \"Single Cell Type RNA - Mucus-secreting cells [NX]\"\n",
    "]\n",
    "\n",
    "hpa_features = hpa.iloc[:, hpa.columns.isin(identifiers+discrete_features+continuous_features)]\n",
    "\n",
    "hpa_features[\"Tissue RNA - lung [NX]\"] = (hpa_features[\"Tissue RNA - lung [NX]\"] - hpa_features[\"Tissue RNA - lung [NX]\"].mean()) / hpa_features[\"Tissue RNA - lung [NX]\"].std()\n",
    "col = \"Single Cell Type RNA - Mucus-secreting cells [NX]\"\n",
    "hpa_features[col] = (hpa_features[col] - hpa_features[col].mean()) / hpa_features[col].std()\n",
    "\n",
    "def explode(feature) :\n",
    "    return feature.apply(lambda x: x.replace(' ', '').split(','))\n",
    "\n",
    "hpa_clean = hpa.fillna('')\n",
    "for ft in discrete_features :\n",
    "    hpa_clean[ft] = explode(hpa_clean[ft])\n",
    "\n",
    "protein_class = hpa_clean[\"Protein class\"].explode().unique()\n",
    "biological_process = hpa_clean[\"Biological process\"].explode().unique()\n",
    "molecular_function = hpa_clean[\"Molecular function\"].explode().unique()\n",
    "disease_involvement = hpa_clean[\"Disease involvement\"].explode().unique()\n",
    "subcellular_location = hpa_clean[\"Subcellular location\"].explode().unique()\n",
    "GO_features = np.concatenate([protein_class, biological_process, molecular_function, disease_involvement, subcellular_location])\n",
    "\n",
    "RowFeatures = pd.DataFrame(data = 0,index = hpa_clean['Ensembl'],columns=GO_features)\n",
    "counter = 0\n",
    "\n",
    "for index, row in RowFeatures.iterrows() :\n",
    "    features = hpa_clean.iloc[counter][['Protein class', 'Biological process', 'Molecular function', 'Disease involvement', 'Subcellular location']].to_list()\n",
    "    flattened = [item for sublist in features for item in sublist if item]\n",
    "    for t in flattened :\n",
    "        row[t] = 1\n",
    "    counter +=1 \n",
    "\n",
    "n_comp = 100\n",
    "svd = TruncatedSVD(n_components = n_comp)\n",
    "svdModel = svd.fit(RowFeatures)\n",
    "visits_emb = svdModel.transform(RowFeatures)\n",
    "hpa = pd.DataFrame(data=visits_emb, index=RowFeatures.index).reset_index().rename(columns={\"Ensembl\":\"ensembl\"})\n",
    "hpa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ensembl</th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>...</th>\n",
       "      <th>95</th>\n",
       "      <th>96</th>\n",
       "      <th>97</th>\n",
       "      <th>98</th>\n",
       "      <th>99</th>\n",
       "      <th>nih_ssm_in_cohort</th>\n",
       "      <th>nih_ssm_across_gdc</th>\n",
       "      <th>nih_cnv_gain</th>\n",
       "      <th>nih_cnv_loss</th>\n",
       "      <th>nih_tot_mutations</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000175899</td>\n",
       "      <td>0.443505</td>\n",
       "      <td>0.313574</td>\n",
       "      <td>0.430307</td>\n",
       "      <td>-0.285337</td>\n",
       "      <td>-0.351412</td>\n",
       "      <td>1.545059</td>\n",
       "      <td>-0.450936</td>\n",
       "      <td>-0.444200</td>\n",
       "      <td>-0.116774</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.079607</td>\n",
       "      <td>0.503267</td>\n",
       "      <td>0.576245</td>\n",
       "      <td>0.444453</td>\n",
       "      <td>-0.574186</td>\n",
       "      <td>0.049383</td>\n",
       "      <td>0.031607</td>\n",
       "      <td>0.042885</td>\n",
       "      <td>0.066277</td>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000128274</td>\n",
       "      <td>0.532289</td>\n",
       "      <td>0.789331</td>\n",
       "      <td>-0.122159</td>\n",
       "      <td>0.019000</td>\n",
       "      <td>1.427162</td>\n",
       "      <td>-0.065659</td>\n",
       "      <td>0.001669</td>\n",
       "      <td>0.386983</td>\n",
       "      <td>-0.028900</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.110665</td>\n",
       "      <td>0.140087</td>\n",
       "      <td>-0.089804</td>\n",
       "      <td>0.000472</td>\n",
       "      <td>0.025348</td>\n",
       "      <td>0.001764</td>\n",
       "      <td>0.009436</td>\n",
       "      <td>0.048733</td>\n",
       "      <td>0.025341</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000094914</td>\n",
       "      <td>1.605229</td>\n",
       "      <td>0.936681</td>\n",
       "      <td>0.345886</td>\n",
       "      <td>0.183699</td>\n",
       "      <td>-1.001764</td>\n",
       "      <td>-0.846346</td>\n",
       "      <td>0.550995</td>\n",
       "      <td>0.368205</td>\n",
       "      <td>0.202731</td>\n",
       "      <td>...</td>\n",
       "      <td>0.027753</td>\n",
       "      <td>-0.048839</td>\n",
       "      <td>-0.054099</td>\n",
       "      <td>-0.007369</td>\n",
       "      <td>0.078670</td>\n",
       "      <td>0.005291</td>\n",
       "      <td>0.008822</td>\n",
       "      <td>0.068226</td>\n",
       "      <td>0.019493</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000081760</td>\n",
       "      <td>0.467196</td>\n",
       "      <td>0.690415</td>\n",
       "      <td>-0.066723</td>\n",
       "      <td>0.335478</td>\n",
       "      <td>1.017124</td>\n",
       "      <td>0.029579</td>\n",
       "      <td>-0.131714</td>\n",
       "      <td>0.212387</td>\n",
       "      <td>0.075120</td>\n",
       "      <td>...</td>\n",
       "      <td>0.010099</td>\n",
       "      <td>0.120493</td>\n",
       "      <td>0.138474</td>\n",
       "      <td>-0.167452</td>\n",
       "      <td>0.051508</td>\n",
       "      <td>0.031746</td>\n",
       "      <td>0.015036</td>\n",
       "      <td>0.038986</td>\n",
       "      <td>0.052632</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000114771</td>\n",
       "      <td>0.956718</td>\n",
       "      <td>0.202132</td>\n",
       "      <td>-0.490047</td>\n",
       "      <td>-0.524460</td>\n",
       "      <td>0.467132</td>\n",
       "      <td>-0.207775</td>\n",
       "      <td>-0.494652</td>\n",
       "      <td>0.192615</td>\n",
       "      <td>0.077836</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.015291</td>\n",
       "      <td>0.011153</td>\n",
       "      <td>-0.007147</td>\n",
       "      <td>-0.007970</td>\n",
       "      <td>0.007936</td>\n",
       "      <td>0.010582</td>\n",
       "      <td>0.010817</td>\n",
       "      <td>0.066277</td>\n",
       "      <td>0.011696</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14893</th>\n",
       "      <td>ENSG00000198455</td>\n",
       "      <td>1.436813</td>\n",
       "      <td>-1.455513</td>\n",
       "      <td>0.510199</td>\n",
       "      <td>0.224208</td>\n",
       "      <td>0.323736</td>\n",
       "      <td>-0.171141</td>\n",
       "      <td>-0.060697</td>\n",
       "      <td>0.088626</td>\n",
       "      <td>0.103071</td>\n",
       "      <td>...</td>\n",
       "      <td>0.011581</td>\n",
       "      <td>-0.021367</td>\n",
       "      <td>0.014229</td>\n",
       "      <td>-0.007050</td>\n",
       "      <td>0.008374</td>\n",
       "      <td>0.024691</td>\n",
       "      <td>0.023168</td>\n",
       "      <td>0.033138</td>\n",
       "      <td>0.011696</td>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14894</th>\n",
       "      <td>ENSG00000070476</td>\n",
       "      <td>1.140170</td>\n",
       "      <td>-1.162868</td>\n",
       "      <td>0.551409</td>\n",
       "      <td>0.127004</td>\n",
       "      <td>0.119115</td>\n",
       "      <td>-0.120434</td>\n",
       "      <td>-0.421260</td>\n",
       "      <td>0.798186</td>\n",
       "      <td>0.041917</td>\n",
       "      <td>...</td>\n",
       "      <td>0.022374</td>\n",
       "      <td>-0.019030</td>\n",
       "      <td>0.024306</td>\n",
       "      <td>-0.011823</td>\n",
       "      <td>0.017133</td>\n",
       "      <td>0.014109</td>\n",
       "      <td>0.013195</td>\n",
       "      <td>0.033138</td>\n",
       "      <td>0.015595</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14895</th>\n",
       "      <td>ENSG00000162378</td>\n",
       "      <td>0.782655</td>\n",
       "      <td>-0.057857</td>\n",
       "      <td>-0.397211</td>\n",
       "      <td>0.017498</td>\n",
       "      <td>-0.135123</td>\n",
       "      <td>-0.248882</td>\n",
       "      <td>-0.438176</td>\n",
       "      <td>0.113363</td>\n",
       "      <td>-0.295955</td>\n",
       "      <td>...</td>\n",
       "      <td>0.049873</td>\n",
       "      <td>0.018236</td>\n",
       "      <td>-0.051592</td>\n",
       "      <td>0.049855</td>\n",
       "      <td>-0.009883</td>\n",
       "      <td>0.022928</td>\n",
       "      <td>0.011891</td>\n",
       "      <td>0.035088</td>\n",
       "      <td>0.048733</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14896</th>\n",
       "      <td>ENSG00000159840</td>\n",
       "      <td>1.042306</td>\n",
       "      <td>0.183392</td>\n",
       "      <td>-0.327697</td>\n",
       "      <td>0.065425</td>\n",
       "      <td>-0.459893</td>\n",
       "      <td>0.507171</td>\n",
       "      <td>-0.610148</td>\n",
       "      <td>0.067066</td>\n",
       "      <td>0.368216</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.002111</td>\n",
       "      <td>-0.092111</td>\n",
       "      <td>-0.032771</td>\n",
       "      <td>0.033117</td>\n",
       "      <td>-0.052496</td>\n",
       "      <td>0.012346</td>\n",
       "      <td>0.011047</td>\n",
       "      <td>0.064327</td>\n",
       "      <td>0.023392</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14897</th>\n",
       "      <td>ENSG00000074755</td>\n",
       "      <td>0.553160</td>\n",
       "      <td>0.068709</td>\n",
       "      <td>0.013208</td>\n",
       "      <td>0.760541</td>\n",
       "      <td>0.592771</td>\n",
       "      <td>-0.094544</td>\n",
       "      <td>0.326136</td>\n",
       "      <td>-0.636445</td>\n",
       "      <td>0.047971</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.017633</td>\n",
       "      <td>-0.008296</td>\n",
       "      <td>0.002019</td>\n",
       "      <td>0.006353</td>\n",
       "      <td>-0.002431</td>\n",
       "      <td>0.044092</td>\n",
       "      <td>0.035827</td>\n",
       "      <td>0.011696</td>\n",
       "      <td>0.046784</td>\n",
       "      <td>26</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>14898 rows × 106 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               ensembl         0         1         2         3         4  \\\n",
       "0      ENSG00000175899  0.443505  0.313574  0.430307 -0.285337 -0.351412   \n",
       "1      ENSG00000128274  0.532289  0.789331 -0.122159  0.019000  1.427162   \n",
       "2      ENSG00000094914  1.605229  0.936681  0.345886  0.183699 -1.001764   \n",
       "3      ENSG00000081760  0.467196  0.690415 -0.066723  0.335478  1.017124   \n",
       "4      ENSG00000114771  0.956718  0.202132 -0.490047 -0.524460  0.467132   \n",
       "...                ...       ...       ...       ...       ...       ...   \n",
       "14893  ENSG00000198455  1.436813 -1.455513  0.510199  0.224208  0.323736   \n",
       "14894  ENSG00000070476  1.140170 -1.162868  0.551409  0.127004  0.119115   \n",
       "14895  ENSG00000162378  0.782655 -0.057857 -0.397211  0.017498 -0.135123   \n",
       "14896  ENSG00000159840  1.042306  0.183392 -0.327697  0.065425 -0.459893   \n",
       "14897  ENSG00000074755  0.553160  0.068709  0.013208  0.760541  0.592771   \n",
       "\n",
       "              5         6         7         8  ...        95        96  \\\n",
       "0      1.545059 -0.450936 -0.444200 -0.116774  ... -0.079607  0.503267   \n",
       "1     -0.065659  0.001669  0.386983 -0.028900  ... -0.110665  0.140087   \n",
       "2     -0.846346  0.550995  0.368205  0.202731  ...  0.027753 -0.048839   \n",
       "3      0.029579 -0.131714  0.212387  0.075120  ...  0.010099  0.120493   \n",
       "4     -0.207775 -0.494652  0.192615  0.077836  ... -0.015291  0.011153   \n",
       "...         ...       ...       ...       ...  ...       ...       ...   \n",
       "14893 -0.171141 -0.060697  0.088626  0.103071  ...  0.011581 -0.021367   \n",
       "14894 -0.120434 -0.421260  0.798186  0.041917  ...  0.022374 -0.019030   \n",
       "14895 -0.248882 -0.438176  0.113363 -0.295955  ...  0.049873  0.018236   \n",
       "14896  0.507171 -0.610148  0.067066  0.368216  ... -0.002111 -0.092111   \n",
       "14897 -0.094544  0.326136 -0.636445  0.047971  ... -0.017633 -0.008296   \n",
       "\n",
       "             97        98        99  nih_ssm_in_cohort  nih_ssm_across_gdc  \\\n",
       "0      0.576245  0.444453 -0.574186           0.049383            0.031607   \n",
       "1     -0.089804  0.000472  0.025348           0.001764            0.009436   \n",
       "2     -0.054099 -0.007369  0.078670           0.005291            0.008822   \n",
       "3      0.138474 -0.167452  0.051508           0.031746            0.015036   \n",
       "4     -0.007147 -0.007970  0.007936           0.010582            0.010817   \n",
       "...         ...       ...       ...                ...                 ...   \n",
       "14893  0.014229 -0.007050  0.008374           0.024691            0.023168   \n",
       "14894  0.024306 -0.011823  0.017133           0.014109            0.013195   \n",
       "14895 -0.051592  0.049855 -0.009883           0.022928            0.011891   \n",
       "14896 -0.032771  0.033117 -0.052496           0.012346            0.011047   \n",
       "14897  0.002019  0.006353 -0.002431           0.044092            0.035827   \n",
       "\n",
       "       nih_cnv_gain  nih_cnv_loss  nih_tot_mutations  \n",
       "0          0.042885      0.066277                 29  \n",
       "1          0.048733      0.025341                  1  \n",
       "2          0.068226      0.019493                  3  \n",
       "3          0.038986      0.052632                 19  \n",
       "4          0.066277      0.011696                  6  \n",
       "...             ...           ...                ...  \n",
       "14893      0.033138      0.011696                 16  \n",
       "14894      0.033138      0.015595                 10  \n",
       "14895      0.035088      0.048733                 14  \n",
       "14896      0.064327      0.023392                  8  \n",
       "14897      0.011696      0.046784                 26  \n",
       "\n",
       "[14898 rows x 106 columns]"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "master = hpa.merge(gdc, on=\"ensembl\")\n",
    "master"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Import STRING Dataset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Original Size:  11938498\n"
     ]
    }
   ],
   "source": [
    "el_map = pd.read_csv('../data/raw_data/9606.protein.aliases.v11.5.txt', sep=\"\\t\")\n",
    "el = pd.read_csv('../data/raw_data/9606.protein.links.detailed.v11.5.txt', sep=\" \")\n",
    "el_map.loc[el_map.source == 'Ensembl_gene']\n",
    "\n",
    "el = remove_redun(el, True)\n",
    "el = map_IDs(el, el_map, verbose=True)\n",
    "\n",
    "el_map"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<bound method Index.unique of Int64Index([      0,       1,       2,       3,       4,       5,       6,\n",
      "                  7,       8,       9,\n",
      "            ...\n",
      "            5969239, 5969240, 5969241, 5969242, 5969243, 5969244, 5969245,\n",
      "            5969246, 5969247, 5969248],\n",
      "           dtype='int64', length=5969249)>\n"
     ]
    }
   ],
   "source": [
    "print(el.index.unique)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Disease Gene Network Merge"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>EI_gda</th>\n",
       "      <th>gda_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NAT2</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ADA</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CDH2</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CDKN2B-AS1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>HOTAIR</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.02</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>978</th>\n",
       "      <td>ZEB2</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.04</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>979</th>\n",
       "      <td>CDC20</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>980</th>\n",
       "      <td>MFN2</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>981</th>\n",
       "      <td>MVP</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.03</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>982</th>\n",
       "      <td>CDH1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.23</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>983 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           Gene  EI_gda  gda_score\n",
       "0          NAT2     1.0       0.02\n",
       "1           ADA     1.0       0.02\n",
       "2          CDH2     1.0       0.02\n",
       "3    CDKN2B-AS1     1.0       0.12\n",
       "4        HOTAIR     1.0       0.02\n",
       "..          ...     ...        ...\n",
       "978        ZEB2     1.0       0.04\n",
       "979       CDC20     1.0       0.03\n",
       "980        MFN2     1.0       0.03\n",
       "981         MVP     1.0       0.03\n",
       "982        CDH1     1.0       0.23\n",
       "\n",
       "[983 rows x 3 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dgn = pd.read_csv(\"../data/raw_data/gda_disease_summary_luad.csv\")\n",
    "dgn_dict = pd.read_csv(\"../data/raw_data/gda_dictionary.csv\")\n",
    "\n",
    "score_threshold = 0.02\n",
    "ei_threshold = 0.7\n",
    "\n",
    "dgn = dgn[['Gene', 'EI_gda', 'Score_gda']]\n",
    "dgn = dgn.loc[dgn['Score_gda'] >= score_threshold]\n",
    "dgn = dgn.loc[dgn['EI_gda'] > ei_threshold]\n",
    "dgn.rename({'Score_gda':'gda_score'}, axis=1, inplace=True)\n",
    "dgn = dgn.merge(dgn_dict, on=\"Gene\").drop(['Unnamed: 0', 'Unnamed: 0.1', 'ensembl'], axis=1)\n",
    "dgn"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "interpreter": {
   "hash": "4410ce168b863aee9b4918bd897f8b9953e347697e7905a6c1ad714c78ed846f"
  },
  "kernelspec": {
   "display_name": "Python 3.8.6 64-bit",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.7"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
